File: petclinic-java-gradle-react-tailwind-ts/src/main/java/dev/pollito/petclinic_java_gradle_react_tailwind_ts/vet/VetRepository.java
package dev.pollito.petclinic_java_gradle_react_tailwind_ts.vet;

import java.util.List;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;

public interface VetRepository extends JpaRepository<Vet, Integer> {

    Page<Vet> findAllById(Integer id, Pageable pageable);

    List<Vet> findAllByVetSpecialtySpecialtiesId(Integer id);

}

File: petclinic-kt-gradle-thymeleaf-tailwind-webpack-htmx-ts/src/main/kotlin/dev/pollito/petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts/vet/VetRepository.kt
package dev.pollito.petclinic_kt_gradle_thymeleaf_tailwind_webpack_htmx_ts.vet

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.data.jpa.repository.JpaRepository


interface VetRepository : JpaRepository<Vet, Int> {

    fun findAllById(id: Int?, pageable: Pageable): Page<Vet>

    fun findAllByVetSpecialtySpecialtiesId(id: Int): List<Vet>

}
